ui/pages/institutional/interactive-replacement-token-page/interactive-replacement-token-page.js

```diff
@@ -0,0 +1,349 @@
+  const {
+    selectedAddress,
+    custodyAccountDetails,
+    interactiveReplacementToken,
+    mmiConfiguration,
+  } = useSelector((state) => state.metamask);
+  const { custodianName } =
+    custodyAccountDetails[toChecksumHexAddress(address || selectedAddress)] ||
+    {};
+  const { url } = interactiveReplacementToken || {};
+  const { custodians } = mmiConfiguration;
+  const custodian =
+    custodians.find((item) => item.name === custodianName) || {};
+  const mostRecentOverviewPage = useSelector(getMostRecentOverviewPage);
+  const metaMaskAccounts = useSelector(getMetaMaskAccounts);
+  const connectRequests = useSelector(
+    (state) => state.metamask.institutionalFeatures?.connectRequests,
+  );
+  const [isLoading, setIsLoading] = useState(false);
+  const [tokenAccounts, setTokenAccounts] = useState([]);
+  const [error, setError] = useState(false);
+  const t = useI18nContext();
+  const [copied, handleCopy] = useCopyToClipboard();
+  const {
+    removeAddTokenConnectRequest,
+    setCustodianNewRefreshToken,
+    getCustodianAccounts,
+  } = mmiActions;
+  const connectRequest = connectRequests ? connectRequests[0] : undefined;
+
+  useEffect(() => {
+    isMountedRef.current = true;
+    return () => (isMountedRef.current = false);
+  }, []);
+
+  useEffect(() => {
+    const getTokenAccounts = async () => {
+      if (!connectRequest) {
+        history.push(mostRecentOverviewPage);
+        return;
+      }
+
+      try {
+        const custodianAccounts = await dispatch(
+          getCustodianAccounts(
+            connectRequest.token,
+            connectRequest.apiUrl,
+            connectRequest.service,
+            false,
+          ),
+        );
+
+        const filteredAccounts = custodianAccounts.filter(
+          (account) => metaMaskAccounts[account.address.toLowerCase()],
+        );
+
+        const mappedAccounts = filteredAccounts.map((account) => ({
+          address: account.address,
+          name: account.name,
+          labels: account.labels,
+          balance:
+            metaMaskAccounts[account.address.toLowerCase()]?.balance || 0,
+        }));
+
+        if (isMountedRef.current) {
+          setTokenAccounts(mappedAccounts);
+          setIsLoading(false);
+        }
+      } catch (e) {
+        console.log(e);
+        setError(true);
+        setIsLoading(false);
+      }
+    };
+
+    getTokenAccounts();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  if (!connectRequest) {
+    history.push(mostRecentOverviewPage);
+    return null;
+  }
+
+  const onRemoveAddTokenConnectRequest = ({ origin, apiUrl, token }) => {
+    dispatch(
+      removeAddTokenConnectRequest({
+        origin,
+        apiUrl,
+        token,
+      }),
+    );
+  };
+
+  const handleReject = () => {
+    onRemoveAddTokenConnectRequest(connectRequest);
+    history.push(mostRecentOverviewPage);
+  };
+
+  const handleApprove = async () => {
```

Comment:
Can you add some test to this action?

Reply:
What type of tests do you want? Testing the logic inside the handleApprove method or what the user will see if they click Approve?

Reply:
You can follow this one:
- render component
- click approve button
- expect `setCustodianNewRefreshToken` mock is called with this object 
```
{
              address: account.address,
              newAuthDetails: {
                refreshToken: connectRequest.token,
                refreshTokenUrl: connectRequest.apiUrl,
              },
            }
```
- expect `showInteractiveReplacementTokenBanner` is called 
- expect `history.push` is called with xxx
- and also it is interesting to assert this component is gone because of the isLoading 
```
          <footer>
            <PulseLoader />
          </footer>
```